zerver/views/push_notifications.py

```diff
@@ -66,3 +67,19 @@ def remove_android_reg_id(
     validate_token(token, PushDeviceToken.GCM)
     remove_push_device_token(user_profile, token, PushDeviceToken.GCM)
     return json_success(request)
+
+
+@human_users_only
+@has_request_variables
+def send_test_push_notification_api(
+    request: HttpRequest, user_profile: UserProfile, token: str = REQ()
+) -> HttpResponse:
+    # TODO: Should we split this endpoint by token kind (apple/android)?
+    # Should we do any extra validation like the other endpoints above?
+    try:
+        device = PushDeviceToken.objects.get(token=token, user=user_profile)
+    except PushDeviceToken.DoesNotExist:
+        raise JsonableError(_("Token does not exist"))
```

Comment:
I'm leaning towards the answer that we don't need to separate endpoints here or `kind`-based validation, because unlike the registration/deregistration endpoints above, we don't create some kind of db record that we would risk having malformed. But perhaps splitting would allow more helpful error responses for notification debugging? Not sure

Reply:
I don't see much reason to split the endpoint by `kind`; though as noted in https://chat.zulip.org/#narrow/stream/378-api-design/topic/.2323997.20Endpoint.20for.20test.20notification/near/1638499, we might want have the `token` parameter be optional to support sending to all devices.

Reply:
Okay, removed the TODO and the "send to all devices" functionality is now implemented too.